src/Microsoft.TestPlatform.CrossPlatEngine/Adapter/TestExecutionRecorder.cs

```diff
@@ -32,6 +38,9 @@ public TestExecutionRecorder(ITestCaseEventsHandler testCaseEventsHandler, ITest
             this.testRunCache = testRunCache;
             this.testCaseEventsHandler = testCaseEventsHandler;
             this.attachmentSets = new List<AttachmentSet>();
+
+            this.testResultDictionary = new Dictionary<Guid, List<TestResult>>();
```

Comment:
Please add a comment explaining why the error correction is required. It will help in future.